modules/sc-mesh-secure-deployment/src/nats/src/bat_ctrl_utils.py

```diff
@@ -0,0 +1,280 @@
+                        self.logger.info(
+                            "Failed to set hop penalty %s for: %s. Error: %s",
+                            penalty,
+                            interface,
+                            e,
+                        )
+
+        # Set hop penalty for hard interfaces
+        hardif_hop_penalty = self.__hop_penalty.get("hardif", {})
+        self.logger.info(f"hardif_hop_penalty: {hardif_hop_penalty}")
+        if hardif_hop_penalty is not None:
+            for interface, penalty in hardif_hop_penalty.items():
+                if penalty is not None:
+                    try:
+                        # Serch hardif from all my batmans
+                        for batman_interface in self.__bat_interfaces:
+                            hardif = self.__find_batman_hardif(interface, batman_interface)
+                            if hardif:
+                                subprocess.run(
+                                    [
+                                        "batctl",
+                                        "hardif",
+                                        hardif,
+                                        "hop_penalty",
+                                        str(penalty),
+                                    ],
+                                    check=True,
+                                )
+                                break
+                    except Exception as e:
+                        self.logger.info(
+                            "Failed to set hop penalty %s for: %s. Error: %s",
+                            penalty,
+                            interface,
+                            e,
+                        )
+
+    @staticmethod
+    def __get_interface_mac(interface: str) -> Union[None, str]:
+        try:
+            with open(
+                file=f"/sys/class/net/{interface}/address",
+                mode="r",
+                encoding="utf-8",
+            ) as file:
+                return file.readline().strip()
+        except FileNotFoundError:
+            return None
+
+    def __find_batman_hardif(self, interface: str, batman_if: str) -> str:
+        # Check if the interface exists in the batctl if list
+        try:
+            batctl_output = subprocess.check_output(["batctl", batman_if, "if"]).decode(
+                "utf-8"
+            )
+            if interface in batctl_output:
+                self.logger.info(
+                    "Interface %s found in %s interface list.", interface, batman_if
+                )
+                return interface
+        except subprocess.CalledProcessError as e:
+            self.logger.warning(
+                "Interface %s not found in %s interface list. Error: %s",
+                interface,
+                batman_if,
+                e,
+            )
+
+        # If interface is not found, try to find it using MAC address
+        mac_address = self.__get_interface_mac(interface).replace(":", "")
```

Comment:
As `__get_interface_mac` can return `None`, you have to move the `replace` inside the `if` below to avoid an exception :package:

Reply:
Agreed

Reply:
Fixed via cc2d5663f8ca5c884be02417f1011a5697087166